x-pack/plugins/apm/public/components/shared/environment_select/index.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { useState, useCallback } from 'react';
+import { debounce } from 'lodash';
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { getEnvironmentLabel } from '../../../../common/environment_filter_values';
+import { SERVICE_ENVIRONMENT } from '../../../../common/elasticsearch_fieldnames';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
+import { Environment } from '../../../../common/environment_rt';
+
+export function EnvironmentSelect({
+  environment,
+  environmentOptions,
+  status,
+  serviceName,
+  start,
+  end,
+  onChange,
+}: {
+  environment: Environment;
+  environmentOptions: Array<EuiComboBoxOptionOption<string>>;
+  status: FETCH_STATUS;
+  serviceName?: string;
+  start?: string;
+  end?: string;
```

Comment:
can we make these required?

Reply:
we can make the ranges required if we set the default values for ranged `start` and `end` in the consuming component. 

Now I have them optional because I get them from the params and handle the a [default values](https://github.com/elastic/kibana/pull/126679/files#diff-d75a3e48748c8c5022fdcb98f1cd9766a9ca02d7af80531f174c09a5480a1b78R62) here but it seems a little bit confusing. 

Which way do you think it will be cleaner. 

- the consuming app sets the default values (24h-now) and passes the ranged to `EnvironmentSelect` which are required
- the `EnvironmentSelect` component should use `useApmParams` to get the ranges and handles the default values accordingly

Reply:
the first option sounds good to me. The consuming component should decide what the time range is because it knows in what context `EnvironmentSelect` is being used.